nupic/algorithms/anomaly.py

```diff
@@ -123,25 +126,9 @@ def compute(self, activeColumns, predictedColumns,
                               	(used in anomaly-likelihood)
     @return the computed anomaly score; float 0..1
     """
-    # Start by computing the raw anomaly score.
-    anomalyScore = computeRawAnomalyScore(activeColumns, predictedColumns)
 
     # Compute final anomaly based on selected mode.
-    if self._mode == Anomaly.MODE_PURE:
-      score = anomalyScore
-    elif self._mode == Anomaly.MODE_LIKELIHOOD:
-      if inputValue is None:
-        raise ValueError("Selected anomaly mode 'Anomaly.MODE_LIKELIHOOD' "
-                 "requires 'inputValue' as parameter to compute() method. ")
-
-      probability = self._likelihood.anomalyProbability(
-          inputValue, anomalyScore, timestamp)
-      # low likelihood -> hi anomaly
-      score = 1 - probability
-    elif self._mode == Anomaly.MODE_WEIGHTED:
-      probability = self._likelihood.anomalyProbability(
-          inputValue, anomalyScore, timestamp)
-      score = anomalyScore * (1 - probability)
+    score = self.getComputeFn(self._mode)(activeColumns, predictedColumns, inputValue, timestamp)
```

Comment:
The code shouldn't be repeating the `getComputeFn` computation every time, since `self._mode` is not changing after instantiation of the class. The computation function should be saved in a member variable during instantiation of the class, instead.